ext_mongodb.php

```diff
@@ -746,7 +868,7 @@ public function getCode() : string
 
 	public function getScope() : mixed
 	{
-		if ( $this->scope !== NULL )
+		if ( isset( $this->scope) && $this->scope !== NULL )
```

Comment:
I don't understand how this related to `__set_state()`.

In PHPC, my get_properties handler (currently used for serialization and debug info) only includes "scope" if it is a BSON document (i.e. assigned from an array or object). There is some code that conditionally sets it to null (PHP value) if the bson-to-PHP conversion fails, but that would accompany an exception.

If "scope" was not initialized from an array or object, I omit it from the hash table returned by get_properties.

My `__set_state()` and unserialize code paths use the same logic discussed above: scope is only assigned if it is an array or object. Otherwise, I leave it uninitialized (i.e. BSON document pointer remains null).